src/utils.js

```diff
@@ -90,3 +91,33 @@ export function valueOf(doc) {
   const { docId, versionId, links, forks, createdAt, updatedAt, ...rest } = doc
   return rest
 }
+
+/**
+ * Create an internal ID from a project key
+ * @param {Buffer} projectKey
+ * @returns {import('./types.js').ProjectId}
+ */
+export function projectKeyToId(projectKey) {
+  return /** @type {import('./types.js').ProjectId} */ (
+    projectKey.toString('hex')
+  )
+}
+
+/**
+ * Create a public ID from a project key
+ * @param {Buffer} projectKey
+ * @returns {import('./types.js').ProjectPublicId}
+ */
+export function projectKeyToPublicId(projectKey) {
+  return /** @type {import('./types.js').ProjectPublicId} */ (
+    keyToPublicId(projectKey)
+  )
+}
+
+/**
+ * @param {import('./types.js').ProjectId} projectId
+ * @returns {Buffer} 24-byte nonce (same length as sodium.crypto_aead_xchacha20poly1305_ietf_NPUBBYTES)
+ */
+export function projectIdToNonce(projectId) {
+  return Buffer.from(projectId, 'hex').subarray(0, 24)
```

Comment:
didn't use the sodium constant because it seems that `sodium-javascript` doesn't export it